src/V3PerPackage/PerProcessProcessor.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.V3PerPackage
+{
+    public class PerProcessProcessor
+    {
+        private readonly PerWorkerProcessor _perWorkerProcessor;
+        private readonly ILogger<PerProcessContext> _logger;
+
+        public PerProcessProcessor(PerWorkerProcessor perWorkerProcessor, ILogger<PerProcessContext> logger)
+        {
+            _perWorkerProcessor = perWorkerProcessor;
+            _logger = logger;
+        }
+
+        public async Task ProcessAsync(PerProcessContext context)
+        {
+            var evenCount = context.MessageCount / context.WorkerCount;
```

Comment:
This division of work is not great because the process could end up in a state where it's waiting for one worker to finish its allotted work. I could improve this by making all of the workers drain a shared `MessageCount` counter. Probably not worth the effort.